### PR TITLE
[SGP-29166] change key in common fields

### PR DESCRIPTION
**Pull request by ecedmondson**
## Provider Specifics
Twitch returns a display_name field in the oauth exchange, but we cannot assume that the display name is a given, legal name of some sort. Assuming such, as extract_common_fields does here, results in the BaseProvider, when creating a sociallogin from the response, parsing the display name as a first name because the key was "name"

### CODE REVIEW

File: allauth/socialaccount/providers/twitch/provider.py [hunk 1]
```diff
@@ -28,7 +28,6 @@ def extract_uid(self, data):
     def extract_common_fields(self, data):
         return {
             "username": data.get("login"),
-            "name": data.get("display_name"),
```
Comment (ecedmondson) [inline allauth/socialaccount/providers/twitch/provider.py:31]: We cannot assume that display_name == some_sort_of_given_legal_name. 
Reference: 
https://dev.twitch.tv/docs/api/reference/#get-users

The ticket also has a screenshot of the types of errors this causes on our Platform. 

I will probably submit an upstream ticket to djanog-allauth but this internal way seemed faster for our customer's upcoming event.

**Comment by ecedmondson** [comment]
So obviously the tests are not successful.....I have never made a PR here before...I will try to see if it is because it is a fork.

**Comment by ecedmondson** [comment]
Oh hmm:
`Error: Version 3.5 with arch x64 not found`

**Comment by rbevers** [comment]
> Oh hmm: `Error: Version 3.5 with arch x64 not found`

Just remove Python 3.5 support, yes?

**Comment by ecedmondson** [comment]
> Just remove Python 3.5 support, yes?

Yes probably...got distracted by other support things. I'll take this back.

**Comment by ecedmondson** [comment]
Look like we let previous PRs fail these testenv tests too....
https://github.com/bevy/django-allauth/pull/10

I tried to remove the outdated Python and Django but the docs and standardjs are also failing, even though they pass locally. I will keep  looking into those failures...but I was wondeirng what your thought were on the testenv jobs failing @rbevers . Normally, i would try to figure that out as well, but the customer is wanting this sooner rather than later. However...if we break our allauth fork to do this, many customers are unhappy.

**Comment by ecedmondson** [comment]
The docs failure might be easy to fix, but this is weird that this failure is only happenign in ci:
`docs: failed with make is not allowed, use allowlist_externals to allow it`

**Comment by ecedmondson** [comment]
I am going to try to do this all in the signup form Platform side to derisk this PR and maybe we can have a convo about what to do about django-allauth

**Comment by ecedmondson** [comment]
Yeah I just did this in the signup form for now due to the customer's deadline, and I will probably bring up django-allauth in our team retro

**Pull request closed without merging**